network/wsNetwork_test.go

```diff
@@ -1381,25 +1375,25 @@ func TestPeeringWithIdentityChallenge(t *testing.T) {
 	assert.Equal(t, 0, len(netB.GetPeers(PeersConnectedOut)))
 	// netA never attempts to set identity as it never sees a verified identity
 	assert.Equal(t, 1, netA.identityTracker.(*mockIdentityTracker).getSetCount())
-	// netB would attempt to add the identity to the tracker
-	// but it would not end up being added
-	assert.Equal(t, 2, netB.identityTracker.(*mockIdentityTracker).getSetCount())
+	// no connecton => netB does attepmt to add the identity to the tracker
+	// and it would not end up being added
+	assert.Equal(t, 1, netB.identityTracker.(*mockIdentityTracker).getSetCount())
 	assert.Equal(t, 1, netB.identityTracker.(*mockIdentityTracker).getInsertCount())
 
 	// Check deduplication again, this time from A
 	// the "ok" from tryConnectReserveAddr is overloaded here because isConnectedTo
 	// will prevent this connection from attempting in the first place
 	// in the real world, that isConnectedTo doesn't always trigger, if the hosts are behind
 	// a load balancer or other NAT
-	if _, ok := netA.tryConnectReserveAddr(addrB); ok || true {
-		netA.wg.Add(1)
-		netA.tryConnect(addrB, gossipB)
-		// let the tryConnect go forward
-		time.Sleep(250 * time.Millisecond)
-	}
+	_, ok = netA.tryConnectReserveAddr(addrB)
+	assert.False(t, ok)
+	netA.wg.Add(1)
+	netA.tryConnect(addrB, gossipB)
+	// let the tryConnect go forward
+	time.Sleep(250 * time.Millisecond)
```

Comment:
Can we do this without a single sleep - perhaps periodic waitUntil type checks with a long interruptible timeout on the checking? Getting these sleep times right is tricky (also slows the build down) and can lead to random failures when the build server is congested.

Reply:
assert.Eventually is what I am thinking of

Reply:
idk, this connection attempt is going to be aborted and `tryConnect` does not return an error. I'll check if I can rely on a counter or a logged message here